### DISCUSSION TITLE
API to Link IDP/Social Login to existing user

### DISCUSSION THREAD

**Discussion by bradleybluebean** [zitadel/zitadel#5860] [Q&A] [answered]
The [Create/Import User API](https://zitadel.com/docs/apis/mgmt/management-service-import-human-user) has the ability to link an Identity Provider. Is there a comparable API to link an IDP to an existing user? I see that you can [remove a link from a User](https://zitadel.com/docs/apis/mgmt/management-service-remove-human-linked-idp), but I don't see a comparable "add link" option.

**Comment by hifabienne** [comment, marked as answer]
Hei @bradleybluebean 
At the moment everything login related has to be done through ZITADELs login UI.
However, we are currently working on our new Login/Register API to make it possible for you to build this parts by yourself.
You can find the correlating issue here: https://github.com/zitadel/zitadel/issues/5015
We have just opened a new github discussion to get some feedback on our first implemented parts: https://github.com/zitadel/zitadel/discussions/5875

**Reply by bradleybluebean** [reply]
Thanks for the update and the links

**Comment by bmphjhu906** [comment]
Hei @hifabienne 
Has this function been completed？: #5016 
How to use the username and password request API to complete login.

**Reply by hifabienne** [reply]
Hei @bmphjhu906 
Login with Username and Password with the new API is already possible.
I am currently writing a guide for our docs with a detailed description.
For now. You can create the session directly with the username and password and it will be checked. Or create an empty session or only with the username, and update it later in the login process:
- Create Session: https://zitadel.com/docs/apis/resources/session_service/session-service-create-session
- Update Session: https://zitadel.com/docs/apis/resources/session_service/session-service-set-session

After each request you will get a session token. 
Note: This are opaque tokens. OIDC Standard is not yet implemented, but will be asap.
